examples/graphbolt/disk_based_feature/README.md

```diff
@@ -0,0 +1,39 @@
+## Overview
+
+This project demonstrates how to use GraphBolt to train and evaluate a GraphSAGE model for node classification task on large graphs, where node features are on-disk and fetched using `DiskBasedFeature`. GraphBolt utilizes [S3-FIFO](https://s3fifo.com) to cache frequently required features and io_uring to fetch cache-missed features from disk.
+
+# Node classification Task
+
+This example demonstrates how to run node classification task with **GraphBolt.DiskBasedFeature**.
+
+## Run on `ogbn-papers100M` dataset
+
+This instruction trains on three-layer GraphSAGE model for 3 epochs on `ogbn-papers100M` dataset with 10GB CPU cache, using neighbor sampling.
+
+* Dataset:
+
+|     Dataset     | Graph Size | Feature Size | Feature Dim |
+| :-------------: | :--------: | :----------: | :---------: |
+| Ogbn-papers100M |   13 GB   |    53 GB    |     128     |
+
+* Instruction:
+
+```
+python node_classification.py --num-gpu-cached-features=0 --num-cpu-cached-features=20000000 --batch-dependency=128 --cpu-feature-cache-policy=s3-fifo --dataset=ogbn-papers100M --sample-mode=sample_neighbor --epochs=3
```

Comment:
```suggestion
python node_classification.py --num-gpu-cached-features=0 --num-cpu-cached-features=20000000 --batch-dependency=128 --dataset=ogbn-papers100M --sample-mode=sample_neighbor --epochs=3
```

Reply:
SIEVE is the default option and seems to perform better. Can you verify?

Reply:
I verified that SIEVE actually performed much better than S3-fifo. The result is impressive:
```
Training: 1178it [03:37,  5.41it/s, num_nodes=674195, gpu_cache_miss=1, cpu_cache_miss=0.0697]
Evaluating: 123it [00:09, 12.46it/s, num_nodes=627374, gpu_cache_miss=1, cpu_cache_miss=0.0661]
Epoch 00, Loss: 1.4129, Approx. Train: 0.5794, Approx. Val: 0.6331, Time: 217.73547530174255s
Training: 1178it [01:32, 12.80it/s, num_nodes=658826, gpu_cache_miss=1, cpu_cache_miss=0.042] 
Evaluating: 123it [00:10, 11.91it/s, num_nodes=627153, gpu_cache_miss=1, cpu_cache_miss=0.0411]
Epoch 01, Loss: 1.1438, Approx. Train: 0.6388, Approx. Val: 0.6363, Time: 92.05023622512817s
Training: 1178it [01:30, 12.98it/s, num_nodes=652847, gpu_cache_miss=1, cpu_cache_miss=0.0327]
Evaluating: 123it [00:10, 11.99it/s, num_nodes=626515, gpu_cache_miss=1, cpu_cache_miss=0.0324]
Epoch 02, Loss: 1.0951, Approx. Train: 0.6519, Approx. Val: 0.6460, Time: 90.72666501998901s
Training: 1178it [01:31, 12.90it/s, num_nodes=666681, gpu_cache_miss=1, cpu_cache_miss=0.0281]
Evaluating: 123it [00:09, 12.65it/s, num_nodes=625135, gpu_cache_miss=1, cpu_cache_miss=0.0279]
Epoch 03, Loss: 1.0671, Approx. Train: 0.6590, Approx. Val: 0.6605, Time: 91.28739047050476s
```

Reply:
Are these results with updated master? Just pushed another optimization.

Reply:
Try using LRU and CLOCK as well. We might want to show the users the difference between different caching policies.

Reply:
Seems there is almost no performance difference using the save_aligned dataset for obgn-papers100M. Following is the comparison on the two versions of papers100M, using SIEVE algorithm:

* Original Feature File:

```
Namespace(epochs=9999999, lr=0.001, num_hidden=256, dropout=0.2, batch_size=1024, num_workers=0, dataset='ogbn-papers100M', fanout='10,10,10', mode='pinned-pinned-cuda', layer_dependency=False, batch_dependency=128, cpu_feature_cache_policy=None, num_cpu_cached_features=20000000, num_gpu_cached_features=0, early_stopping_patience=25, sample_mode='sample_neighbor', precision='high', enable_inference=False)

Training: 1178it [04:17,  4.57it/s, num_nodes=669981, gpu_cache_miss=1, cpu_cache_miss=0.0825]
Evaluating: 123it [00:17,  6.87it/s, num_nodes=626529, gpu_cache_miss=1, cpu_cache_miss=0.0799]
Epoch 00, Loss: 1.4106, Approx. Train: 0.5796, Approx. Val: 0.6212, Time: 257.7615864276886s
Training: 1178it [01:52, 10.43it/s, num_nodes=656458, gpu_cache_miss=1, cpu_cache_miss=0.0585]
Evaluating: 123it [00:15,  7.76it/s, num_nodes=627928, gpu_cache_miss=1, cpu_cache_miss=0.0581]
Epoch 01, Loss: 1.1438, Approx. Train: 0.6391, Approx. Val: 0.6435, Time: 112.89340138435364s
Training: 1178it [01:52, 10.45it/s, num_nodes=669632, gpu_cache_miss=1, cpu_cache_miss=0.0505]
Evaluating: 123it [00:15,  8.01it/s, num_nodes=625727, gpu_cache_miss=1, cpu_cache_miss=0.0504]
Epoch 02, Loss: 1.0954, Approx. Train: 0.6519, Approx. Val: 0.6550, Time: 112.74339509010315s
```

* Using `gb.numpy_save_aligned` Feature File:

```
Namespace(epochs=9999999, lr=0.001, num_hidden=256, dropout=0.2, batch_size=1024, num_workers=0, dataset='ogbn-papers100M', fanout='10,10,10', mode='pinned-pinned-cuda', layer_dependency=False, batch_dependency=128, cpu_feature_cache_policy=None, num_cpu_cached_features=20000000, num_gpu_cached_features=0, early_stopping_patience=25, sample_mode='sample_neighbor', precision='high', enable_inference=False)

Training: 1178it [04:17,  4.58it/s, num_nodes=662186, gpu_cache_miss=1, cpu_cache_miss=0.0825]
Evaluating: 123it [00:17,  6.87it/s, num_nodes=624853, gpu_cache_miss=1, cpu_cache_miss=0.0799]
Epoch 00, Loss: 1.4156, Approx. Train: 0.5783, Approx. Val: 0.6181, Time: 257.26628851890564s
Training: 1178it [01:53, 10.42it/s, num_nodes=655680, gpu_cache_miss=1, cpu_cache_miss=0.0585]
Evaluating: 123it [00:15,  7.76it/s, num_nodes=626433, gpu_cache_miss=1, cpu_cache_miss=0.058] 
Epoch 01, Loss: 1.1451, Approx. Train: 0.6385, Approx. Val: 0.6344, Time: 113.09730005264282s
Training: 1178it [01:53, 10.42it/s, num_nodes=656730, gpu_cache_miss=1, cpu_cache_miss=0.0505]
Evaluating: 123it [00:15,  8.00it/s, num_nodes=626782, gpu_cache_miss=1, cpu_cache_miss=0.0504]
Epoch 02, Loss: 1.0968, Approx. Train: 0.6512, Approx. Val: 0.6632, Time: 113.00190234184265s
```

So I was thinking, maybe it is not necessary to update the datasets in S3 using save_aligned version for papers100M and those smaller than it as there is no observable difference.